core/src/main/java/com/novoda/noplayer/internal/mediaplayer/AndroidMediaPlayerImpl.java

```diff
@@ -255,12 +259,20 @@ public void attach(PlayerView playerView) {
 
     @Override
     public void detach(PlayerView playerView) {
+        clearSurfaceHolderCallbacks();
         surfaceHolderRequester = null;
```

Comment:
I wasn't sure about it, as it meant the method was doing 2 things and kinda hid the nulling. 

I would expect a detach method to null stuff up. I'm easy though